contracts/extensions/votingReputation/VotingReputation.sol

```diff
@@ -472,29 +266,36 @@ contract VotingReputation is ColonyExtension, BasicMetaTransaction, VotingReputa
       motion.votes[NAY] < motion.votes[YAY]
     );
 
-    if ((
-        getSig(motion.action) == SET_EXPENDITURE_STATE ||
-        getSig(motion.action) == SET_EXPENDITURE_PAYOUT
-      ) && getTarget(motion.altTarget) == address(colony)
-    ) {
-      bytes32 structHash = hashExpenditureActionStruct(motion.action);
-      expenditureMotionCounts[structHash] -= 1;
-
-      // Release the claimDelay if this is the last active motion
-      if (expenditureMotionCounts[structHash] == 0) {
-        bytes memory claimDelayAction = createClaimDelayAction(motion.action, 0);
-        // No require this time, since we don't want stakes to be permanently locked
-        executeCall(_motionId, claimDelayAction);
+    if (motion.sig == bytes4(0)) { // Backwards compatibility for versions 9 and below
```

Comment:
Hmm. That's valid I suppose, although I expect this would be an extremely unlikely case. Let me think about it for a bit.

Reply:
Hmm. In our case, the only times this branching is relevant is if we're dealing with an expenditure-related action. So we have a few scenarios:

- An old motion is created with a function chosen to have a 0x0 signature. In that case, we wouldn't enter this branch because it's not an expenditure-related function.
- A new motion is created with a function chosen to have a 0x0 signature. Same logic applies.
- A multicall is created with an expenditure function and the final function chosen to have a 0x0 signature. In that case, the sig will be the expenditure signature.

I don't see a scenario where an expenditure-related motion could end up with a sig value of 0x0 which isn't an upgrade, and if someone attempted to engineer a motion to have such a signature, these branches would not be hit. What do you think? It's possible I've made a mistake in my analysis.

Reply:
I think I agree with this analysis, but I still don't think I'm comfortable with this. In general with this PR, there's a fair amount of weird behaviour, or edge cases like this one that don't _currently_ matter, but by it existing it becomes far too easy to introduce bugs down the line, as well as more likely there being a bug that we've not spotted, so I'd like to make things  as consistent as possible.

In terms of cleanly identifying 'old' motions, wee could store `motionCount` when the contract was updated to v10, and key based off the combination of that and `motionId` to determine whether a motion needs the 'old' functionality or not?

Reply:
Sure. I'll re-iterate that I think this extension should be sunset in the near future and replaced with a V2, but until then I'm happy to implement the backwards-compatibility in this way.